src/library_syscall.js

```diff
@@ -1839,7 +1839,8 @@ for (var x in SyscallsLibrary) {
   if (x in WASI_SYSCALLS) {
     wasi = true;
   } else {
-    var match = /^__sys_[^_]*$/.exec(x);
+    // A syscall is __sys_X (but at least one syscall has an extra _ prefix)
+    var match = /^__sys__?[^_]*$/.exec(x);
```

Comment:
Can we just replace this with something like `if (!x.startswith('__sys_') { continue }`   Do we even need a regex here?

Reply:
I found a way to simplify this I think, PTAL. It also has to check if it's not `__proxy` etc. which is what was guarded against before.